test/images/sample-apiserver/Dockerfile

```diff
@@ -0,0 +1,33 @@
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FROM BASEIMAGE
+
+RUN apk add --no-cache git musl-dev go
```

Comment:
this seems pretty heavyweight for an e2e test image. can we build the binary separately and then copy it into a lighter-weight image?

Reply:
Ack, i'll throw in a multi-stage build.